### PR TITLE
Add openssl_publickey_info module

### PR DESCRIPTION
**Pull request by felixfontein**
##### SUMMARY
Also share code between openssl_privatekey_info and the new module, and improve documentation of it.

Prepares #227.

##### ISSUE TYPE
- New Module Pull Request

##### COMPONENT NAME
openssl_publickey_info
openssl_privatekey_info

### CODE REVIEW

File: plugins/modules/openssl_privatekey_info.py [hunk 1]
```diff
@@ -130,6 +130,62 @@
...
+            returned: When C(type=RSA)
+        exponent:
+            description:
+                - The RSA key's public exponent.
+            type: int
+            returned: When C(type=RSA)
+        p:
+            description:
+                - The C(p) value for DSA.
+                - This is the prime modulo which arithmetic takes place.
```
Comment (Ajpantuso) [inline plugins/modules/openssl_privatekey_info.py:152]:
```suggestion
                - This is the prime modulus upon which arithmetic takes place.
```
Modulo is an operation; Modulus is an operand.

File: plugins/modules/openssl_privatekey_info.py [hunk 2]
```diff
@@ -130,6 +130,62 @@
...
+            type: int
+            returned: When C(type=DSA)
+        q:
+            description:
+                - The C(q) value for DSA.
+                - This is a prime that divices C(p - 1), and at the same time the order of the subgroup of the
```
Comment (Ajpantuso) [inline plugins/modules/openssl_privatekey_info.py:158]:
```suggestion
                - This is a prime that divides C(p - 1), and at the same time the order of the subgroup of the
```

File: plugins/modules/openssl_publickey_info.py [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+            returned: When C(type=RSA)
+        exponent:
+            description:
+                - The RSA key's public exponent.
+            type: int
+            returned: When C(type=RSA)
+        p:
+            description:
+                - The C(p) value for DSA.
+                - This is the prime modulo which arithmetic takes place.
```
Comment (Ajpantuso) [inline plugins/modules/openssl_publickey_info.py:119]:
```suggestion
                - This is the prime modulus upon which arithmetic takes place.
```

File: plugins/modules/openssl_publickey_info.py [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+            type: int
+            returned: When C(type=DSA)
+        q:
+            description:
+                - The C(q) value for DSA.
+                - This is a prime that divices C(p - 1), and at the same time the order of the subgroup of the
```
Comment (Ajpantuso) [inline plugins/modules/openssl_publickey_info.py:125]:
```suggestion
                - This is a prime that divides C(p - 1), and at the same time the order of the subgroup of the
```

**Review by Ajpantuso — CHANGES_REQUESTED** [review]
A couple of trivial doc updates, but don't see any functional or code quality issues.

File: plugins/modules/openssl_privatekey_info.py (hunk 1 above)
Comment (felixfontein) [reply plugins/modules/openssl_privatekey_info.py:152]: Heh. I should actually know that! Thanks for spotting :)

**Review by Ajpantuso — APPROVED** [review]
LGTM, thanks!

**Comment by felixfontein** [comment]
@Ajpantuso thanks a lot for reviewing this!

**Pull request merged**